remote/fastapi/fileservice/core.py

```diff
@@ -0,0 +1,41 @@
+# Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import pathlib
+
+from .models import FileTree, FileType
+
+logger = logging.getLogger("core")
+
+
+def get_files(dir: pathlib.PosixPath, show_hidden: bool) -> FileTree:
+    file = FileTree(
+        path=str(dir.absolute()),
+        name=dir.name,
+        type=FileType.DIRECTORY,
+        children=[],
+    )
+
+    assert isinstance(file.children, list)
+
+    for item in dir.iterdir():
+        if not show_hidden and item.name.startswith("."):
+            continue
+        if item.is_dir():
+            file.children.append(get_files(item, show_hidden))
+        elif item.is_file():
+            file.children.append(
+                FileTree(
+                    name=item.name,
+                    path=str(item.absolute()),
+                    type=FileType.FILE,
+                )
+            )
+        else:
+            logger.info(
+                "Unsupported file type for file %s. Only directory and file are supported.",
+                str(file),
+            )
```

Comment:
It's not `str(file)` whose file type isn't supported, it's `item`.

And I think it's better to write item name at the end of a log statement to make the name more clear. Maybe:
"There is no supported file type (file or directory) for file: %s"